custom_components/stromer/stromer.py

```
@@ -170,7 +170,23 @@ async def stromer_get_code(self) -> None:
         next_loc = res.headers.get("Location")
+        if not next_loc:
+            LOGGER.debug("No next location returned from Stromer API. Full response details:")
+            LOGGER.debug("  **DONT DOX BY SHARING** Full Request-Data: %s", data)
+            LOGGER.debug("  **SHARE RESPONSIBLY ** Full Partial-Data: %s", data["next"])
+            LOGGER.debug("  Status: %s", res.status)
+            LOGGER.debug("  Headers: %s", dict(res.headers))
+            try:
+                body_text = await res.text()
+                LOGGER.debug("  Body: %s", body_text)
+            except Exception as err:
+                raise NextLocationError("Unable to provide body information from Stromer API") from err
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)